OpenRA.Platforms.Default/OpenGL.cs

```diff
@@ -553,5 +662,46 @@ public static void WriteGraphicsLog(string message)
 			for (var i = 0; i < extensionCount; i++)
 				Log.Write("graphics", glGetStringi(GL_EXTENSIONS, (uint)i));
 		}
+
+		public static void CheckGLError()
+		{
+			// Let the debug message handler log the errors instead.
+			if (Features.HasFlag(GLFeatures.DebugMessagesCallback))
+				return;
+
+			var type = glGetError();
+			if (type == GL_NO_ERROR)
+				return;
+
+			var errorText = ErrorToText[type];
+			var error = "GL Error: {0}\n{1}".F(errorText, new StackTrace());
+
+			WriteGraphicsLog(error);
+
+			const string exceptionMessage = "OpenGL Error: See graphics.log for details.";
+
+			if (type == GL_OUT_OF_MEMORY)
+				throw new OutOfMemoryException(exceptionMessage);
+
+			throw new InvalidOperationException(exceptionMessage);
+		}
+
+		static void DebugMessageHandler(int source, int type, uint id, int severity, int length, StringBuilder message, IntPtr userparam)
+		{
+			if (severity != GL_DEBUG_SEVERITY_HIGH &&
+			    severity != GL_DEBUG_SEVERITY_MEDIUM)
```

Comment:
finding and fixing all the potential performance warnings under different drivers is probably not feasible, so I suggest we print `GL_DEBUG_SEVERITY_MEDIUM` to stdout but don't write the graphics log or throw.